src/otx/core/data/entity/base.py

```diff
@@ -669,25 +669,29 @@ class OTXBatchPredEntity(OTXBatchDataEntity):
 
     Attributes:
         scores: List of probability scores representing model predictions.
-        saliency_maps: List of saliency maps used to explain model predictions.
+        saliency_map: List of saliency maps used to explain model predictions.
             This field is optional and will be an empty list for non-XAI pipelines.
-        feature_vectors: List of intermediate feature vectors used for model predictions.
+        feature_vector: List of intermediate feature vectors used for model predictions.
             This field is optional and will be an empty list for non-XAI pipelines.
     """
 
     scores: list[np.ndarray] | list[Tensor]
 
     # (Optional) XAI-related outputs
-    saliency_maps: list[np.ndarray] | list[Tensor] = field(default_factory=list)
-    feature_vectors: list[np.ndarray] | list[Tensor] = field(default_factory=list)
+    # TODO(vinnamkim): These are actually plural, but their namings are not
+    # This is because ModelAPI requires the OV IR to produce `saliency_map`
+    # and `feature_vector` (singular) named outputs.
+    # It should be fixed later.
+    saliency_map: list[np.ndarray] | list[Tensor] = field(default_factory=list)
+    feature_vector: list[np.ndarray] | list[Tensor] = field(default_factory=list)
```

Comment:
Do you want to make updates in Model API and then rename `saliency_map -> saliency_maps` back to plural?

Reply:
Yes, `feature_vector` would be too.